atc/worker/gardenruntime/worker.go

```diff
@@ -382,55 +386,42 @@ type mountableLocalInput struct {
 	mountPath string
 }
 
-type remoteInput struct {
-	volume    runtime.Artifact
-	mountPath string
-}
-
 func (worker *Worker) cloneInputVolumes(
 	ctx context.Context,
 	spec runtime.ContainerSpec,
 	privileged bool,
 	container db.CreatingContainer,
 	delegate runtime.BuildStepDelegate,
 ) ([]runtime.VolumeMount, map[string]bool, error) {
+
 	inputDestinationPaths := make(map[string]bool)
+	localInputs := make([]mountableLocalInput, len(spec.Inputs))
+	g, groupCtx := errgroup.WithContext(ctx)
 
-	var localInputs []mountableLocalInput
-	var remoteInputs []remoteInput
+	ctx, span := tracing.StartSpan(ctx, "worker.cloneInputVolumes", tracing.Attrs{"container_id": container.Handle()})
+	defer span.End()
 
-	for _, input := range spec.Inputs {
-		volume, ok, err := worker.findVolumeForArtifact(ctx, spec.TeamID, input.Artifact)
-		if err != nil {
-			return nil, nil, err
-		}
+	for i, input := range spec.Inputs {
+		i, input := i, input // keep a local copy for the goroutine
 
 		cleanedInputPath := filepath.Clean(input.DestinationPath)
 		inputDestinationPaths[cleanedInputPath] = true
 
-		if ok && volume.DBVolume().WorkerName() == worker.Name() {
-			localInputs = append(localInputs, mountableLocalInput{
-				cowParent: volume.(Volume),
-				mountPath: input.DestinationPath,
-			})
-		} else {
-			remoteInputs = append(remoteInputs, remoteInput{
-				volume:    input.Artifact,
+		g.Go(func() error {
```

Comment:
Do we need to pass in the `input`/`i` variables into the argument of the goroutines? Just so that it will get the correct value within every loop.

```suggestion
		g.Go(func(i int, input []runtime.Input) error {
```

Reply:
@clarafu This is not needed. If you look at a few lines back, `i` and `input` have been made as local copy within the `for` loop. But anyway, I have changed it to a more comprehensive implementation in #8322.